sql/src/main/java/io/crate/cluster/gracefulstop/DecommissioningService.java

```diff
@@ -323,9 +323,27 @@ private void setGracefulStopTimeout(TimeValue gracefulStopTimeout) {
 
     private void setGracefulStopForce(boolean forceStop) {
         this.forceStop = forceStop;
+        if (!exitActionProvided) {
+            setExitAction();
+        }
     }
 
     private void setDataAvailability(DataAvailability dataAvailability) {
         this.dataAvailability = dataAvailability;
+        if (!exitActionProvided) {
```

Comment:
Don't we have to run this all the time so that the delay is updated if the data availability setting is changed at runtime?